static/src/javascripts/projects/common/modules/article/rich-links.js

```diff
@@ -81,8 +81,11 @@ define([
             !isSensitive &&
             !isDuplicate
         ) {
+            if (typeof richLinkTagTmpl === 'string') {
+                richLinkTagTmpl = template(richLinkTagTmpl);
+            }
```

Comment:
Variable re-assignments hurt code readability. Here we're not sure if `richLinkTagTmpl` is a string or something else. I think creating a new variable here (e.g. `richLinkTagTmplString` and `richLinkTagTmpl`) would be better, to make it explicit that there are two types. (On a side note, I wonder if there's an ESLint rule for this?)

Ideally I think we should pre-compile templates as an optimisation at the module loader level, like you said. I would worry that, if we added a half-way solution such as this, we would never get around to doing The Right Thing™. How hard would it be to do that?